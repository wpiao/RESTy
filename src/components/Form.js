import React from 'react';

const Form = ({ handleSubmit, handleChange, handleClick, url, handleGo }) => {
  return (
    <form onSubmit={handleSubmit} >
      <input type="text" onChange={handleChange} value={url} />
      <button value="get" onClick={handleClick} >GET</button>
      <button value="post" onClick={handleClick} >POST</button>
      <button value="put" onClick={handleClick} >PUT</button>
      <button value="patch" onClick={handleClick} >PATCH</button>
      <button value="delete" onClick={handleClick} >DELETE</button>
      <input type="submit" onClick={handleGo} value="Go!" />
    </form>
  )
}

export default Form;