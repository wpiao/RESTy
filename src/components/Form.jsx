import React from 'react';

const Form = ({ handleSubmit, handleChange, handleClick, url }) => {
  return (
    <form onSubmit={handleSubmit} >
      <h1 className="header">RESTy</h1>
      <input type="text" onChange={handleChange} value={url} />
      <button value="get" onClick={handleClick} >GET</button>
      <button value="post" onClick={handleClick} >POST</button>
      <button value="put" onClick={handleClick} >PUT</button>
      <button value="patch" onClick={handleClick} >PATCH</button>
      <button value="delete" onClick={handleClick} >DELETE</button>
      <input type="submit" value="Go!" />
    </form>
  )
}

export default Form;