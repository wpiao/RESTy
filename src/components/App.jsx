import React from 'react';
import Form from './Form.jsx';
const axios = require('axios');

class App extends React.Component {
  constructor(props) {
    super(props);
    this.state = {
      method: 'get',
      url: '',
      count: 0,
      results: []
    };
  }

  // add some functions
  handleChange = e => {
    const url = e.target.value;
    this.setState({ url });
  }

  handleClick = e => {
    e.preventDefault();
    const method = e.target.value;
    this.setState({ method });
  }

  handleSubmit = e => {
    e.preventDefault();
    axios({
      method: this.state.method,
      url: this.state.url
    })
      .then(response => console.log(response))
  }

  render() {
    const { url } = this.state;
    return (
      <React.Fragment>
        <Form url={url} handleSubmit={this.handleSubmit} handleChange={this.handleChange} handleClick={this.handleClick} />
      </React.Fragment>
    )
  }
}

export default App;