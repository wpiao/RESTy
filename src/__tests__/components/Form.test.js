import React from 'react';
import { shallow, mount } from 'enzyme';
import renderer from 'react-test-renderer';
import Form from '../../components/Form.js';
import App from '../../components/App.jsx';

describe('<Form />', () => {
  it('exists after my application has loaded', () => {
    const form = shallow(<Form />);
    expect(form.find('form').exists()).toEqual(true);
    expect(form.find('h1').exists()).toEqual(true);
    expect(form.find('button').exists()).toEqual(true);
    expect(form.find('input').exists()).toEqual(true);
  });

  it('should render correctly', () => {
    const form = renderer.create(<Form />).toJSON();
    expect(form).toMatchSnapshot();
  });

});

