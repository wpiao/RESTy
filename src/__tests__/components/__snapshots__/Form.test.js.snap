// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Form /> should render correctly 1`] = `
<form>
  <h1
    className="header"
  >
    RESTy
  </h1>
  <input
    type="text"
  />
  <button
    value="get"
  >
    GET
  </button>
  <button
    value="post"
  >
    POST
  </button>
  <button
    value="put"
  >
    PUT
  </button>
  <button
    value="patch"
  >
    PATCH
  </button>
  <button
    value="delete"
  >
    DELETE
  </button>
  <input
    type="submit"
    value="Go!"
  />
</form>
`;
